cpp/bench/ann/CMakeLists.txt

```diff
@@ -95,10 +128,10 @@ function(ConfigureAnnBench)
     ${BENCH_NAME}
     PRIVATE raft::raft
             nlohmann_json::nlohmann_json
-            $<$<BOOL:${RAFT_ANN_BENCH_USE_MULTIGPU}>:NCCL::NCCL>
+            $<$<BOOL:${GPU_BUILD}>:$<$<BOOL:${RAFT_ANN_BENCH_USE_MULTIGPU}>:NCCL::NCCL>>
```

Comment:
Question: does anything in ann-bench depend on NCCL at the moment? Maybe we can just remove this line altogether?